plutus-core-spec/figures/PlutusCoreGrammar.tex

```diff
@@ -5,45 +5,47 @@
-                                  &        &     & \lam{x}{S}{M}              & \textrm{$\lambda$ abstraction}\\
+                                  &        &     & \lam{x}{A}{M}              & \textrm{$\lambda$ abstraction}\\
                                   &        &     & \app{M}{N}                 & \textrm{function application}\\
-                                  &        &     & \con{cn}                   & \textrm{builtin}\\
-                                  &        &     & \error{S}                  & \textrm{error}\\
-        \textrm{Value}            &U, V    & ::= & \abs{\alpha}{K}{V}         & \textrm{type abstraction}\\
-                                  &        &     & \wrap{\alpha}{S}{V}        & \textrm{fix type's wrap}\\
-                                  &        &     & \lam{x}{S}{M}              & \textrm{$\lambda$ abstraction}\\
-                                  &        &     & BV                         & \textrm{builtin value}\\
-        \textrm{Builtin Value}    & BV     & ::= & \con{cn}                   & \textrm{builtin}\\
-                                  &        &     & \app{BV}{V}                & \textrm{function application}\\
-                                  &        &     & \inst{BV}{A}               & \textrm{type instantiation}\\
+                                  &        &     & \builtin{bn}{A^*}{M^*}     & \textrm{builtin}\\
+                                  &        &     & \error{A}                  & \textrm{error}\\
+        \textrm{Value}            & U, V   & ::= & \con{cn}                   & \textrm{contant}\\
+                                  &        &     & \abs{\alpha}{K}{V}         & \textrm{type abstraction}\\
+                                  &        &     & \wrap{\alpha}{A}{V}        & \textrm{fix type's wrap}\\
+                                  &        &     & \lam{x}{A}{M}              & \textrm{$\lambda$ abstraction}\\
         \textrm{Type}             & A,B,C  & ::= & \alpha                     & \textrm{type variable}\\
+                                  &        &     & s                          & \textrm{size}\\
                                   &        &     & \funT{A}{B}                & \textrm{function type}\\
                                   &        &     & \allT{\alpha}{K}{A}        & \textrm{polymorphic type}\\
                                   &        &     & \fixT{\alpha}{A}           & \textrm{fixed point type}\\
                                   &        &     & \lamT{\alpha}{K}{A}        & \textrm{$\lambda$ abstraction}\\
                                   &        &     & \appT{A}{B}                & \textrm{function application}\\
-                                  &        &     & \conT{tcn}                 & \textrm{builtin type}\\
-        \textrm{Type Value}       & R,S,T  & ::= & \funT{S}{T}                & \textrm{function type}\\
+                                  &        &     & \conT{tcn}{A}          & \textrm{type constant}\\
```

Comment:
From the previous thread:

> Oh! I've reorgged the way builtins and constants work, so now type constants are just a name and a size (the A arg). I would have added a signature component for them too but all type constants take size args only. If you think having a signature system would be better, let me know.

I do not have any opinion here right now (that's a "being ad hoc vs overcomplicating" dilemma), but if we're going to generalize built-ins (that `dyn` type variable and stuff), this might be not that relevant now.

Reply:
I'm not sure what you mean?

Reply:
I mean, there is a dilemma, but it's probably not important now. I just wanted to make a note which can be of more relevance later.